### PR TITLE
Add 0.5 compat for bytestring changes

### PR DESCRIPTION
**Pull request by quinnj**

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -485,6 +501,8 @@ function _compat(ex::Expr)
             end
         elseif VERSION < v"0.4.0-dev+4389" && f === :withenv
             rewrite_pairs_to_tuples!(ex)
+        elseif VERSION < v"0.4.0" && f == :String
```
Comment (tkelman) [inline src/Compat.jl]: what does this correspond to, the String => AbstractString rename?

Comment (yuyichao) [inline src/Compat.jl]: Code that use `@compat` but not updated for the abstract string renaming can be broken by this.

Comment (quinnj) [inline src/Compat.jl]: This is for 0.3 compatibility since `call` overload wasn't available. @yuyichao, do I need to change something here?

Comment (yuyichao) [inline src/Compat.jl]: .... I forgot that you already confirmed that `bytestring` should be good enough on 0.3 so this should be fine then. https://github.com/JuliaLang/Compat.jl/pull/203#discussion_r64138813

Comment (tkelman) [inline src/Compat.jl]: what is the cutoff version number though?

**Comment by quinnj** [comment]
Good to merge?

File: src/Compat.jl (hunk 1 above)
Comment (quinnj) [inline src/Compat.jl]: Ah, got it. It would just be the same as we use above. I'll update.

**Comment by quinnj** [comment]
Any other comments?

**Comment by nalimilan** [comment]
Would be good to merge https://github.com/JuliaLang/Compat.jl/pull/208 too before tagging a new release.

File: src/Compat.jl [hunk 2]
```diff
@@ -4,12 +4,38 @@ module Compat
...
     typealias ASCIIString Core.String
 else
     typealias String Base.ByteString
+    if VERSION >= v"0.4.0-dev+1246"
+        call(::Type{Base.ByteString},s::Cstring) = bytestring(s::Cstring)
+        call(::Type{Base.ByteString},v::Vector{UInt8}) = bytestring(v::Vector{UInt8})
+        call(::Type{Base.ByteString},io::Base.AbstractIOBuffer) = bytestring(io::Base.AbstractIOBuffer)
+        call(::Type{Base.ByteString},p::Union{Ptr{Int8},Ptr{UInt8}}) = bytestring(p::Union{Ptr{Int8},Ptr{UInt8}})
+        call(::Type{Base.ByteString},p::Union{Ptr{Int8},Ptr{UInt8}}, len::Integer) = bytestring(p::Union{Ptr{Int8},Ptr{UInt8}}, len::Integer)
+        call(::Type{Base.ByteString},s...) = bytestring(s::AbstractString...)
```
Comment (stevengj) [inline src/Compat.jl]: This method does not seem to have been tested... see #210

**Pull request merged**